tests/checkpoint/test_checkpoint.py

```diff
@@ -465,15 +467,18 @@ def test_checkpoint_deserialization_failure(
 
         assert expected_error in str(e.value)
 
-    @pytest.mark.unit
-    def test_checkpoint_deserialization_with_actions(self, mocker: MockerFixture):
-        # Arrange
+    def _set_context(self, mocker: MockerFixture):
```

Comment:
This is what we were doing before but it looks dangerous since it sets some global variable in `set_context(context)` and never undoes it. All our tests should be setting this themselves if needed so maybe this is fine but it feels like we should do this as a context manager and undo it to it's previous state.

Reply:
Made a fixture